pandas/core/indexes/range.py

```diff
@@ -627,18 +628,12 @@ def _evaluate_numeric_binop(self, other):
 
                     return result
 
-                except (ValueError, TypeError, AttributeError):
-                    pass
-
-                # convert to Int64Index ops
-                if isinstance(self, RangeIndex):
-                    self = self.values
-                if isinstance(other, RangeIndex):
-                    other = other.values
-
-                with np.errstate(all='ignore'):
-                    results = op(self, other)
-                return Index(results, **attrs)
+                except (ValueError, TypeError, AttributeError,
+                        ZeroDivisionError):
```

Comment:
when does this raise ZeroDivisionError? isn't he point of the errstate to catch that?

Reply:
>when does this raise ZeroDivisionError?

ATM `pd.RangeIndex(0, 5) / 0` raises `ZeroDivisionError`.

> isn't he point of the errstate to catch that?

I _think_ errstate is to suppress warnings, but not really sure.

Reply:
hmm, that seems wrong, but I guess can cover later